sklearn/meson.build

```diff
@@ -18,6 +18,38 @@ if is_mingw
   add_project_arguments('-mlong-double-64', language: 'c')
 endif
 
+if not py.version().version_compare('>=3.9')
+  error('scikit-learn requires Python>=3.9, got ' + py.version() + ' instead')
+endif
+cython_min_version = run_command(py, ['_min_dependencies.py', 'cython'], check: true).stdout().strip()
+if not cython.version().version_compare('>=' + cython_min_version)
+  error('scikit-learn requires Cython>=' + cython_min_version + ', got ' + cython.version() + ' instead')
+endif
+
+# Only check numpy and scipy versions when not cross-compiling, as numpy and
+# scipy are not installed in the build machine
+if not meson.is_cross_build()
```

Comment:
Note to myself: I think this one should be removed, there are other places in the code where we execute Python script without protection.

I have to say I don't remember all the details about this and need to have a closer look.

Reply:
So I looked a little bit more at this.

It looks like conda-forge does not care about whether there is the protection `if not meson.is_cross_build()`, I tested this in https://github.com/conda-forge/scikit-learn-feedstock/pull/268.

On the other hand, other cross-compilation settings may care more, so I would still leave this just in case but not 100% sure.

For example the scipy Nix package is using `numpy-include-dir` which is also meant to be used for tricky cross-compilation settings (basically you can not run the Python interpreter or the wrong numpy include dir gets picked): https://github.com/NixOS/nixpkgs/blob/81c768e223876ce8ff5120934c5d8f485731c6ed/pkgs/development/python-modules/scipy/default.nix#L61

The scikit-learn Nix package has a patch to get the version directly rather than running a Python script:
https://github.com/NixOS/nixpkgs/blob/fcf65eb2954fc78232b038ceef5d6a5390a734ec/pkgs/development/python-modules/scikit-learn/default.nix#L42-L44